### PR TITLE
Inventory system

### PR DESCRIPTION
**Pull request by GameWyrm**
Added an inventory in the Ship Log by using the [Custom Ship Log Modes mod](https://outerwildsmods.com/mods/customshiplogmodes/) by dgarro. Includes support for reading hints.

### CODE REVIEW

File: mod/ArchConsoleManager.cs
Comment (Ixrec) [inline mod/ArchConsoleManager.cs:142]: should this be `maxLocations`?

File: mod/InGameTracker/TrackerDescriptions.cs
Comment (Ixrec) [inline mod/InGameTracker/TrackerDescriptions.cs:23]: I think some of the "item" variables are getting confusing and requiring lots of comments because they're mixing up multiple concepts of "item".

How do you feel about `itemID` in this function becoming `trackerEntryID`, and then `discoveredItem` being something like `isAnAPItem`? Then the comment could be more like "Because we want to group signals together into one entry per frequency, there isn't a 1-to-1 mapping between AP items and tracker inventory entries".

File: mod/InGameTracker/TrackerInventoryMode.cs
Comment (Ixrec) [inline mod/InGameTracker/TrackerInventoryMode.cs:111]: Wouldn't `tex` be for a texture, not a sprite?

File: mod/InGameTracker/TrackerInventoryMode.cs
Comment (Ixrec) [inline mod/InGameTracker/TrackerInventoryMode.cs:114]: Looks like we're doing the same check and branch here that the descriptions file does. Maybe "descriptions" wasn't the optimal scope for that other file? I'm not too worried about this one causing bugs, so up to you if we try to fix this duplication in this PR, or leave it for when one of us feels like it.

File: mod/InGameTracker/TrackerManager.cs
Comment (Ixrec) [inline mod/InGameTracker/TrackerManager.cs:70]: Maybe `ItemIsNew`? The name `NewItems` had me expecting a list or set of string ids, not an id->bool map.

**Review by Ixrec — COMMENTED** [review]
Looks good to me. A few nits, but I'm happy to merge this and start iterating with it on main.

The only "big" problem I have is all the special casing of the OWV frequency, but that's probably best fixed after a merge so I can extract the frequency<->signal mapping (or figure out where the base game exposes it), and use it everywhere it's needed all at once (note to self: that's at least TrackerDescriptions::DisplayItemText and TrackerManager::MarkItemAsNew/GetFrequency). Plus, I may end up adding similar non-item entries for groups of useful/filler items as I create them, which pushes me more towards "merge then iterate".

File: mod/ArchipelagoRandomizer.csproj.user
Comment (GameWyrm) [reply mod/ArchipelagoRandomizer.csproj.user:9]: Oh my guess is that this happened when I was trying to get that Newtonsoft stuff to work, can be reverted

File: mod/ArchConsoleManager.cs
Comment (GameWyrm) [reply mod/ArchConsoleManager.cs:142]: Yes, it should be

File: mod/InGameTracker/TrackerDescriptions.cs
Comment (GameWyrm) [reply mod/InGameTracker/TrackerDescriptions.cs:23]: I might even go and redo how I do the Tracker Item entries; currently it pulls data from a few different Dictionaries. Might be better to just combine all this data into one struct.

File: mod/InGameTracker/TrackerInventoryMode.cs
Comment (GameWyrm) [reply mod/InGameTracker/TrackerInventoryMode.cs:111]: Yes but originally this returned a Texture2D and then I realized I needed to change the types to a sprite for them to work, and I just didn't bother renaming the variables I had.

**Comment by Ixrec** [comment]
Ran this code on my machine (locally rebased atop my other 0.2.0 changes so far) and everything seems to be working.

File: mod/InGameTracker/TrackerInventoryMode.cs
Comment (GameWyrm) [reply mod/InGameTracker/TrackerInventoryMode.cs:114]: Yeah this could probably be refactored when I change the ItemEntries to use structs.

File: mod/InGameTracker/TrackerManager.cs
Comment (GameWyrm) [reply mod/InGameTracker/TrackerManager.cs:70]: Sure. This will be one of the Dictionaries removed when I change how the ItemEntries work.

**Comment by Ixrec** [comment]
Just to make sure we don't forget: The `maxLocations` used for the progress bar here is / will become incorrect after logsanity, since it's currently counting all locaations in the save file, but the save file records all _possible_ locations (because that's far simpler to implement) including the ones not in this particular multiworld. Personally I'd drop it from this PR and make a separate PR if/when you get it working a different way, but it's up to you.

File: mod/ArchConsoleManager.cs
Comment (Ixrec) [inline mod/ArchConsoleManager.cs:40]: This seems like it would break if the player switched profiles / changed connections, since IIUC we create ArchConsoleManager once near the top of Randomizer.Start() and never recreate it?

(still think the progress indicator might be better as a separate PR since the actual inventory stuff doesn't have this concern)

File: mod/InGameTracker/HowToAddNewItemsToTracker.md
Comment (Ixrec) [inline mod/InGameTracker/HowToAddNewItemsToTracker.md:5]: should this be
```suggestion
public readonly Dictionary<string, InventoryItemEntry> ItemEntries = new()
```
?

File: mod/InGameTracker/TrackerInventoryMode.cs
Comment (Ixrec) [inline mod/InGameTracker/TrackerInventoryMode.cs:63]: dead code

File: mod/ArchConsoleManager.cs
Comment (Ixrec) [reply mod/ArchConsoleManager.cs:40]: It occurs to me that I probably don't _need_ to recreate the `APSession` object on every connection attempt. But that is how it currently works, and I'd definitely want to do manual tests to ensure that subscribing to MutliClient stuff during one connection still works after reconnections using that same session, before we rely on it for code like this. Which is probably not something we want to block this PR on.

**Review by Ixrec — COMMENTED** [review]
The rest LGTM for initial integration. Especially if we do split the location count to a separate PR, the inventory tracker is basically ready to merge IMO. Of course we'll want to add the new items to it before release, but I'm not done adding new items yet so I'll likely want to do that in one big batch at the end.

File: mod/InGameTracker/HowToAddNewItemsToTracker.md
Comment (GameWyrm) [reply mod/InGameTracker/HowToAddNewItemsToTracker.md:5]: Yeah whoops

File: mod/Randomizer.cs
Comment (Ixrec) [inline mod/Randomizer.cs:59]:
```suggestion
        /// The bool is true if the session was manually closed, false if the session was closed via the application closing.
```

**Review by Ixrec — APPROVED** [review]

File: mod/InGameTracker/TrackerManager.cs
Comment (Ixrec) [inline mod/InGameTracker/TrackerManager.cs:7]: LOL this happened to you too

why does this specific include keep spontaneously manifesting?

**Pull request merged**
